File: ScalaComplex/src/complex/PolComplex.scala
package complex

class PolComplex(val radius: Double, val azimuth: Angle) extends AbsComplex {

  def this(real: Double, imag: Double) { this(Conversion.radius(real, imag), Conversion.azimuth(real, imag)) }

  override val real: Double = Conversion.real(radius, azimuth)
  override val imag: Double = Conversion.imag(radius, azimuth)

  override def create(real: Double, imag: Double): Complex = new PolComplex(real, imag)
  override def create(radius: Double, azimuth: Angle): Complex = new PolComplex(radius, azimuth)
}

File: TP1/ComplexJava/src/complex/PolComplex.java
package complex;

public class PolComplex extends AbsComplex {
	private double radius;
	private Angle azimuth;

	public PolComplex(double real, double imag){
		this.radius = Conversion.radius(real, imag);
		this.azimuth = Conversion.azimuth(real, imag);
	}
	public PolComplex(double radius, Angle azimuth){
		this.radius = radius;
		this.azimuth = azimuth;
	}

	public double getReal(){
		return Conversion.real(this.getRadius(), this.getAzimuth());
	}
	public double getImag(){
		return Conversion.imag(this.getRadius(), this.getAzimuth());
	}
	public double getRadius(){
		return radius;
	}
	public Angle getAzimuth(){
		return azimuth;
	}

	public Complex create(double real, double imag){
		return new PolComplex(real, imag);
	}
	public Complex create(double radius, Angle azimuth){
		return new PolComplex(radius, azimuth);
	}
}